apps/spruce/src/pages/waterfall/BuildVariantContext.tsx

```diff
@@ -0,0 +1,44 @@
+import { useContext, createContext, useMemo, useState } from "react";
+
+interface BuildVariantState {
+  columnWidth: number;
+  setColumnWidth: (c: number) => void;
+}
+
+const BuildVariantContext = createContext<BuildVariantState | null>(null);
+
+interface BuildVariantProviderProps {
+  children: React.ReactNode;
+}
+
+const BuildVariantProvider: React.FC<BuildVariantProviderProps> = ({
+  children,
+}) => {
+  const [columnWidth, setColumnWidth] = useState(0);
```

Comment:
Maybe a silly question but how does width come into play here? I would think height is the focus but obviously what you're doing is working... worth leaving a comment perhaps!